src/common/error/src/status_code.rs

```diff
@@ -299,6 +299,27 @@ pub fn status_to_tonic_code(status_code: StatusCode) -> Code {
     }
 }
 
+/// Converts tonic [Code] to [StatusCode].
+pub fn tonic_code_to_status_code(code: Code) -> StatusCode {
+    match code {
+        Code::Ok => StatusCode::Success,
+        Code::Unknown => StatusCode::Unknown,
+        Code::PermissionDenied => StatusCode::PermissionDenied,
+        Code::ResourceExhausted => StatusCode::RateLimited,
+        Code::InvalidArgument | Code::FailedPrecondition | Code::OutOfRange => {
+            StatusCode::InvalidArguments
+        }
+        Code::DeadlineExceeded | Code::Cancelled | Code::Aborted => StatusCode::Cancelled,
+        Code::NotFound
+        | Code::Unavailable
+        | Code::AlreadyExists
+        | Code::Internal
+        | Code::Unauthenticated
+        | Code::DataLoss => StatusCode::Internal,
```

Comment:
Why do we map notfound, alreadyexists, unauthenticated to internal?

Reply:
Changed in [f42e582](https://github.com/GreptimeTeam/greptimedb/pull/5851/commits/f42e58247665244c3a928200ecc200a7eb3a547e)